app/controllers/api/public/v1/providers_controller.rb

```diff
@@ -26,18 +26,43 @@ def updated_since
           @updated_since ||= params.dig(:filter, :updated_since)
         end
 
+        def provider_types
+          return [] if params.dig(:filter, :provider_type).blank?
+          return [] unless params.dig(:filter, :provider_type).is_a?(String)
+
+          params.dig(:filter, :provider_type).split(",")
+        end
+
+        def region_codes
+          return [] if params.dig(:filter, :region_code).blank?
+          return [] unless params.dig(:filter, :region_code).is_a?(String)
+
+          params.dig(:filter, :region_code).split(",")
+        end
+
+        def can_sponsor_skilled_worker_visa?
+          @can_sponsor_skilled_worker_visa ||= params.dig(:filter, :can_sponsor_skilled_worker_visa)&.to_s&.downcase == "true"
+        end
+
+        def can_sponsor_student_visa?
+          @can_sponsor_student_visa ||= params.dig(:filter, :can_sponsor_student_visa)&.to_s&.downcase == "true"
+        end
+
         def providers
           @providers = recruitment_cycle.providers
+
+          @providers = @providers.changed_since(updated_since) if updated_since.present?
```

Comment:
Seeing this makes me wonder if we should introduce a `ProviderSearchService`? All of these methods could probably go in there eg `ProviderSearchService.call(scope:, params:)` or something along those lines and we could clean up the controller quite a bit

Reply:
Here is my previous opinion `CourseSearchService` and the it entailments

https://github.com/DFE-Digital/publish-teacher-training/pull/1435#discussion_r442450179

Reply:
Yup I agree. I still think we can't leave this all in the controller if it builds up. We'll need to put it somewhere. We could learn from the CourseSearchService and do a much better approach (in a future PR once we need to add to this) ie smaller classes which are composed together.